javascript/ql/src/semmle/javascript/Classes.qll

```diff
@@ -153,26 +163,57 @@ class ClassDefinition extends @classdefinition, ClassOrInterface, AST::ValueNode
    */
   override predicate isAbstract() { isAbstractClass(this) }
 
-  override string describe() {
-    if exists(inferNameFromVarDef())
-    then result = inferNameFromVarDef()
-    else result = "anonymous class"
+  override string describe() { inferDescription(result, _) }
+
+  override string getInferredName() {
+    inferDescription(_, result) and
+    result != ""
+  }
+
+  /**
+   * Infers a `description` and `name` of this class.
+   *
+   * If no name could be found, `name` is an empty string.
+   */
+  private predicate inferDescription(string description, string name) {
+    if exists(inferNameFromVarDef(_))
+    then description = inferNameFromVarDef(name)
+    else
+      if exists(inferNameFromProp(_))
+      then description = inferNameFromProp(name)
+      else (
+        description = "anonymous class" and name = ""
+      )
   }
 
   /**
    * Gets a description of this class, based on its declared name or the name
    * of the variable it is assigned to, if any.
    */
-  private string inferNameFromVarDef() {
+  private string inferNameFromVarDef(string name) {
     // in ambiguous cases like `let C = class D {}`, prefer `D` to `C`
```

Comment:
Rather than deciding on a conflict resolution for each ambiguous case, how about we just use both names?